### PR TITLE
Add description generation to LaravelEnumType

### PR DESCRIPTION
**Pull request by jeremynikolic**
⚠️  Waiting for new getClassDescription from [Bensampo\Enum](https://github.com/BenSampo/laravel-enum/pull/270#event-6917233630) to be released ⚠️ 

- [x] Added or updated tests
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

**Changes**

Adjust LaravelEnumType to include an enum description when constructing EnumType.
Makes use of new Bensamp/Enum/Enum::getClassDescription class method

**Breaking changes**

None

### CODE REVIEW

File: src/Schema/Types/LaravelEnumType.php [hunk 1]
```diff
@@ -106,6 +107,16 @@ protected function deprecationReason(Enum $enum): ?string
         return null;
     }
 
+    /**
```
Comment (spawnia) [inline src/Schema/Types/LaravelEnumType.php:110]:
```suggestion
    /**
     * TODO remove check and inline when requiring bensampo/laravel-enum:6
```

File: tests/Unit/Schema/Types/LaravelEnumTypeTest.php [hunk 2]
```diff
@@ -34,6 +34,13 @@ public function testMakeEnumWithCustomName(): void
         $this->assertSame($customName, $enumType->name);
     }
 
+    public function testEnumDescription(): void
+    {
+        $enumType = new LaravelEnumType(LocalizedUserType::class);
+
+        $this->assertSame('Localized user type', $enumType->config['description']);
```
Comment (spawnia) [inline tests/Unit/Schema/Types/LaravelEnumTypeTest.php:41]: Will not work in every case in the test matrix, also needs to check if `getClassDescription` exists and assert accordingly

**Review by spawnia — CHANGES_REQUESTED** [review]

File: tests/Unit/Schema/Types/LaravelEnumTypeTest.php (hunk 2 above)
Comment (jeremynikolic) [reply tests/Unit/Schema/Types/LaravelEnumTypeTest.php:41]: Thanks, I was wondering how to deal with it in this case 👍

File: docs/master/the-basics/types.md [hunk 3]
```diff
@@ -170,6 +170,7 @@ you can use `Nuwave\Lighthouse\Schema\Types\LaravelEnumType` to construct an enu
 
 ```php
 use BenSampo\Enum\Enum;
+use BenSampo\Enum\Attributes\Description;
```
Comment (spawnia) [inline docs/master/the-basics/types.md:173]:
```suggestion
```

File: src/Schema/Types/LaravelEnumType.php [hunk 4]
```diff
@@ -106,6 +107,18 @@ protected function deprecationReason(Enum $enum): ?string
...
+     * TODO remove check and inline when requiring bensampo/laravel-enum:6
+     *
+     * @param  class-string<\BenSampo\Enum\Enum>  $enumClass
+     */
+    protected function enumDescription(string $enumClass): ?string
+    {
+        return method_exists($enumClass, 'getClassDescription')
+            ? $enumClass::getClassDescription()
```
Comment (spawnia) [inline src/Schema/Types/LaravelEnumType.php:118]:
```suggestion
            // @phpstan-ignore-next-line proven to exist by the line above
            ? $enumClass::getClassDescription()
```

File: tests/Unit/Schema/Types/LaravelEnumTypeTest.php [hunk 5]
```diff
@@ -34,6 +34,18 @@ public function testMakeEnumWithCustomName(): void
         $this->assertSame($customName, $enumType->name);
     }
 
+    public function testEnumDescription(): void
+    {
+        $enumType = new LaravelEnumType(LocalizedUserType::class);
+
+        // TODO remove check when requiring bensampo/laravel-enum:6
+        if (method_exists(LocalizedUserType::class, 'getClassDescription')) {
```
Comment (spawnia) [inline tests/Unit/Schema/Types/LaravelEnumTypeTest.php:42]:
```suggestion
        // @phpstan-ignore-next-line depends on the required version
        if (method_exists(LocalizedUserType::class, 'getClassDescription')) {
```

File: tests/Unit/Schema/Types/LaravelEnumTypeTest.php [hunk 6]
```diff
@@ -56,10 +71,24 @@ enum PartiallyDeprecated {
...
+  """Not"""
+  NOT
+
+  """Deprecated"""
+  DEPRECATED @deprecated
+
+  """Deprecated with reason"""
+  DEPRECATED_WITH_REASON @deprecated(reason: "some reason")
+}
+GRAPHQL,
```
Comment (spawnia) [inline tests/Unit/Schema/Types/LaravelEnumTypeTest.php:88]: Trailing comma on the same line is not allowed in PHP 7.2

**Review by spawnia — CHANGES_REQUESTED** [review]

File: docs/master/the-basics/types.md (hunk 3 above)
Comment (spawnia) [reply docs/master/the-basics/types.md:173]: Still unnecessarily present here

File: docs/5/the-basics/types.md [hunk 7]
```diff
@@ -170,7 +170,6 @@ you can use `Nuwave\Lighthouse\Schema\Types\LaravelEnumType` to construct an enu
 
 ```php
 use BenSampo\Enum\Enum;
-use Nuwave\Lighthouse\Schema\Types\LaravelEnumType;
```
Comment (spawnia) [inline docs/5/the-basics/types.md:173]: Necessary. Also, do not change v5 docs, I will do that once merging the PR.

**Review by spawnia — CHANGES_REQUESTED** [review]

**Comment by spawnia** [comment]
You can ignore the errors relating to traits, but https://github.com/nuwave/lighthouse/runs/7213547114?check_suite_focus=true#step:9:76 is still a problem.

File: docs/5/the-basics/types.md [hunk 8]
```diff
@@ -170,7 +170,6 @@ you can use `Nuwave\Lighthouse\Schema\Types\LaravelEnumType` to construct an enu
 
 ```php
 use BenSampo\Enum\Enum;
```
Comment (spawnia) [inline docs/5/the-basics/types.md:172]:
```suggestion
use BenSampo\Enum\Enum;
use Nuwave\Lighthouse\Schema\Types\LaravelEnumType;
```

**Pull request merged**